cmd/node_agent/na/aws.go

```diff
@@ -15,12 +15,39 @@
 package na
 
 import (
+	"crypto/x509"
+	"encoding/base64"
 	"encoding/json"
 	"fmt"
 
 	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/fullsailor/pkcs7"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/credentials"
+	"istio.io/auth/pkg/pki"
+)
+
+const (
+	// AWSCertificatePem is the official public certificate for AWS
+	// copied from https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-identity-documents.html
+	AWSCertificatePem = `-----BEGIN CERTIFICATE-----
```

Comment:
Could you put this content into a file like cert_aws.pem in testdata/? Thanks.

Reply:
but it's not just used for test; it'll be used in production env to verify signature

Reply:
You are right. Maybe put it in a file in a new dir data/?

Reply:
But wouldn't it require the data folder to be deployed alongside the Go binary for deployment?

Reply:
Yes, but will that be a problem?

Reply:
Anyway, we can change this later.

Reply:
My concern is that it kind of goes against the deployment model of Go (take a binary and deploy it any where). If putting the cert in the code is not desired, we can add a cmd option for this.
The deep issue is that AWS doesn't provide a way for us to dynamically retrieve the public cert, nor does it state whether the cert would be rotated in the future.

Reply:
Thanks for the explanation.